django/core/serializers/python.py

```diff
@@ -34,8 +34,32 @@ def end_object(self, obj):
 
     def get_dump_object(self, obj):
         data = {"model": str(obj._meta)}
-        if not self.use_natural_primary_keys or not hasattr(obj, "natural_key"):
+
+        pk_included = True
+
+        if self.use_natural_primary_keys:
+            natural_key_func = getattr(obj, "natural_key", None)
+
+            if callable(natural_key_func):
+                natural_key_value = None
+                try:
+                    natural_key_value = natural_key_func()
+                except Exception:
+                    pass
+
+                is_opt_out = (
+                    natural_key_value is None
+                    or not natural_key_value
+                    or natural_key_value == (obj.pk,)
+                    or not isinstance(natural_key_value, tuple)
+                )
```

Comment:
This handling a lot of variance. Can we just use implicit booleanness? (`not natural_key_value`). This whole block will likely become shorter.